arch/arc/core/cache.c

```diff
@@ -218,12 +218,13 @@ int arch_icache_flush_and_invd_range(void *addr, size_t size)
 
 static int init_dcache(void)
 {
-
-	arch_dcache_enable();
+	if (IS_ENABLED(CONFIG_DCACHE_AUTO_ENABLE)) {
+		arch_dcache_enable();
 
 #if defined(CONFIG_DCACHE_LINE_SIZE_DETECT)
-	init_dcache_line_size();
+		init_dcache_line_size();
```

Comment:
Please don't make this dependent on CONFIG_DCACHE_AUTO_ENABLE. Determining the size should be done during init as it is not done later. The 'build config' register used should be available also with disabled cache, no need to first enable. Other maintenance functions assume caches are enabled, though, which may not be the case if 'enable' has not been called before.

Reply:
> Please don't make this dependent on CONFIG_DCACHE_AUTO_ENABLE. Determining the size should be done during init as it is not done later. The 'build config' register used should be available also with disabled cache, no need to first enable. Other maintenance functions assume caches are enabled, though, which may not be the case if 'enable' has not been called before.

Okay, I will move this out of the if-statement to ensure it is called anyway.